### PR TITLE
[finishes #106013924] Don't allow duplicate element ids in design mode.

### PR DESCRIPTION
**Pull request by davidsbailey**
fixes https://www.pivotaltracker.com/story/show/106013924

### CODE REVIEW

**Comment by davidsbailey** [comment]
Also partially fixes https://www.pivotaltracker.com/story/show/103544734

File: apps/src/applab/designElements/PropertyRow.jsx [hunk 1]
```diff
@@ -61,7 +94,8 @@ var PropertyRow = React.createClass({
         type={this.props.isNumber ? 'number' : undefined}
         value={this.state.value}
         onChange={this.handleChangeInternal}
-        style={rowStyle.input} />;
+        onBlur={this.props.isIdRow ? this.onIdRowBlur : null}
```
Comment (Bjvanminnen) [inline apps/src/applab/designElements/PropertyRow.jsx:97]: Other option would be to always call this.onBlur and have it exit immediately if !this.props.isIdRow. Probably a six of one half dozen of the other sort of thing.

**Comment by Bjvanminnen** [comment]
What's going to happen if I create a button called screen2, then add a new screen? I assume/believe the add screen logic is going to be smart enough to see that screen2 is in use and call the new one screen3, right?

File: apps/test/solutions/applab/ec_design.js [hunk 2]
```diff
@@ -500,6 +500,104 @@ module.exports = {
...
+        ReactTestUtils.Simulate.change(idInput, { target: { value: 'screen1' } });
+        assert.equal(targetScreen.id, "design_screen2", "target screen still has id 'design_screen2'");
+        assert.equal(idInput.style.backgroundColor, "rgb(255, 204, 204)", "duplicate id input 'screen1' has light red background color");
+
+        Applab.onPuzzleComplete();
+      },
+      expected: {
+        result: true,
+        testResult: TestResults.FREE_PLAY
+      }
```
Comment (Bjvanminnen) [inline apps/test/solutions/applab/ec_design.js:600]: nice testing :)

**Comment by Bjvanminnen** [comment]
lgtm

**Comment by davidsbailey** [comment]
correct, we increment the element ID number if there's any conflicting ID in the design mode workspace. That case is kind of tested by this commit from a previous PR: https://github.com/code-dot-org/code-dot-org/commit/6210bf116e351327b80730f0d053e87ded5cd876

File: apps/src/applab/designElements/PropertyRow.jsx (hunk 1 above)
Comment (davidsbailey) [inline apps/src/applab/designElements/PropertyRow.jsx:97]: I thought this way led to slightly better separation of id-specific logic, but I agree it's pretty much a toss-up.

Comment (Bjvanminnen) [inline apps/src/applab/designElements/PropertyRow.jsx:97]: The argument I would make in favor of the other approach (and I don't feel super strongly about it) is that limiting the amount of logic that happens in the JSX makes it easier to quickly scan the component and understand what the output (to the DOM) is.

**Pull request merged**